drivers/pinctrl/Kconfig.gd32

```diff
@@ -6,15 +6,15 @@ DT_COMPAT_GIGADEVICE_GD32_PINCTRL_AFIO := gd,gd32-pinctrl-afio
 
 config PINCTRL_GD32_AF
 	bool "GD32 AF pin controller driver"
-	depends on SOC_FAMILY_GD32 && GD32_HAS_AF_PINMUX
```

Comment:
please, create a new family, e.g. `SOC_FAMILY_GD32V`. Peripherals depend on family/ies. Even though GD32VF103 embeds a Nuclei core, we need to think about it as an SoC (core + peripherals).